### ISSUE TITLE
admissions controller using deprecated API for webhooks

### ISSUE THREAD

**Issue by Robert-Christensen-visa** [volcano-sh/volcano#2025]
**What happened**:

docker desktop has updated their version of Kubernetes to v1.22.  This release of Kubernetes has several deprications, including [admissionregistration](https://kubernetes.io/docs/reference/using-api/deprecation-guide/#webhook-resources-v122)

Seeing this in the logs for the `volcano-admission` controller.  It is not able to register the webhooks.

```
I0216 20:46:02.512254       1 server.go:60] loadAdmissionConf:[]
I0216 20:46:02.513264       1 server.go:78] Registered '/jobs/mutate' as webhook.
I0216 20:46:02.513300       1 server.go:81] Registered configuration for webhook </jobs/mutate>
I0216 20:46:02.513313       1 util.go:53] The service of webhook manager is <volcano-admission-service/volcano-system//jobs/mutate>.
E0216 20:46:02.521788       1 util.go:64] Failed to register mutating admission webhook (/jobs/mutate): the server could not find the requested resource
I0216 20:46:02.522008       1 server.go:78] Registered '/jobs/validate' as webhook.
I0216 20:46:02.522078       1 server.go:81] Registered configuration for webhook </jobs/validate>
I0216 20:46:02.522157       1 util.go:53] The service of webhook manager is <volcano-admission-service/volcano-system//jobs/validate>.
E0216 20:46:02.526002       1 util.go:78] Failed to register validating admission webhook (/jobs/validate): the server could not find the requested resource
I0216 20:46:02.526033       1 server.go:78] Registered '/podgroups/mutate' as webhook.
I0216 20:46:02.526057       1 server.go:81] Registered configuration for webhook </podgroups/mutate>
I0216 20:46:02.526080       1 util.go:53] The service of webhook manager is <volcano-admission-service/volcano-system//podgroups/mutate>.
E0216 20:46:02.528872       1 util.go:64] Failed to register mutating admission webhook (/podgroups/mutate): the server could not find the requested resource
I0216 20:46:02.528904       1 server.go:78] Registered '/pods/validate' as webhook.
I0216 20:46:02.528917       1 server.go:81] Registered configuration for webhook </pods/validate>
I0216 20:46:02.528930       1 util.go:53] The service of webhook manager is <volcano-admission-service/volcano-system//pods/validate>.
E0216 20:46:02.531765       1 util.go:78] Failed to register validating admission webhook (/pods/validate): the server could not find the requested resource
I0216 20:46:02.531799       1 server.go:78] Registered '/pods/mutate' as webhook.
I0216 20:46:02.531809       1 server.go:81] Registered configuration for webhook </pods/mutate>
I0216 20:46:02.531843       1 util.go:53] The service of webhook manager is <volcano-admission-service/volcano-system//pods/mutate>.
E0216 20:46:02.535611       1 util.go:64] Failed to register mutating admission webhook (/pods/mutate): the server could not find the requested resource
I0216 20:46:02.535643       1 server.go:78] Registered '/queues/mutate' as webhook.
I0216 20:46:02.535655       1 server.go:81] Registered configuration for webhook </queues/mutate>
I0216 20:46:02.535668       1 util.go:53] The service of webhook manager is <volcano-admission-service/volcano-system//queues/mutate>.
E0216 20:46:02.538554       1 util.go:64] Failed to register mutating admission webhook (/queues/mutate): the server could not find the requested resource
I0216 20:46:02.538607       1 server.go:78] Registered '/queues/validate' as webhook.
I0216 20:46:02.538621       1 server.go:81] Registered configuration for webhook </queues/validate>
I0216 20:46:02.538635       1 util.go:53] The service of webhook manager is <volcano-admission-service/volcano-system//queues/validate>.
E0216 20:46:02.542371       1 util.go:78] Failed to register validating admission webhook (/queues/validate): the server could not find the requested resource
```

The code that registers the webhooks is using a deprecated interface:

https://github.com/volcano-sh/volcano/blob/eb1fc7f8f102745282718847ffcdd84e28b661f0/cmd/webhook-manager/app/util.go#L37-L41

This will mostly be an issue those starting volcano for the first time in their cluster.  I _think_ once those webhooks are installed restarting the pod will report these errors but since the webhooks will already be registered in Kubernetes the controller will continue to get requests as long as they were already registered.  If volcano was installed in an older version of Kubernetes and Kubernetes is updated it may not cause a change in functionality.

To fix this, the updated API should be used.

**What you expected to happen**:

I expect the controller will be able to register callbacks successfully.

**How to reproduce it (as minimally and precisely as possible)**:

When running the admission controller on Kubernetes >=1.22, these errors are reported.  I have verified this issue on Kubernetes 1.22 and 1.23 using volcano v1.5.0 and the latest build from master branch.

**Anything else we need to know?**:

**Environment**:
- Volcano Version: latest (including v1.5.0)
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.5", GitCommit:"5c99e2ac2ff9a3c549d9ca665e7bc05a3e18f07e", GitTreeState:"clean", BuildDate:"2021-12-16T08:38:33Z", GoVersion:"go1.16.12", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.5", GitCommit:"5c99e2ac2ff9a3c549d9ca665e7bc05a3e18f07e", GitTreeState:"clean", BuildDate:"2021-12-16T08:32:32Z", GoVersion:"go1.16.12", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**Comment by Thor-wl** [comment]
@Robert-Christensen-visa Thanks, Robert, this report is meaningful. Let me take a fix.
/assign

**Comment by william-wang** [comment]
@Robert-Christensen-visa The issue is fixed in #2064, please help to have a try :)

**Comment by Robert-Christensen-visa** [comment]
Thanks @william-wang
